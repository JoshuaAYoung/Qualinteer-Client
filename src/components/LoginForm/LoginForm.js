import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import QualinteerContext from '../../context/QualinteerContext'

export default class LoginForm extends Component {
  static contextType = QualinteerContext;

  demoLogin = (userType) => {
    this.context.setUser(userType);
    userType === 'company'
      ? this.props.history.push('/')
      : this.props.history.push('/jobboard')
  }

  render() {
    return (
      //MAKE SURE TO CHANGE THE BUTTON TYPE BACK TO SUBMIT AND ADD AN EVENT HANDLER TO THE FORM
      <div>
        <h2 className='formHeader'>Log In</h2>
        <h3>TO LOG IN AS A TEST USER:</h3>
        <p>Either click on the "Sign In Company" or "Sign In Volunteer" buttons below to login as one of the user types.</p>
        <p>User registration is disabled for now, though I could use some feedback on the form if you wouldn't mind checking out the "Sign Up" page if you have a moment.</p>
        <p>If you've tested one user type and want to try out the other one, click on the logout link in the navigation bar and come back to this page to sign is as the other user type.</p>
        <form className='signin-form'>
          <div>
            <label htmlFor="email">Email Address</label>
            <input placeholder='Email' type="text" name='email' id='email' />
          </div>
          <div>
            <label htmlFor="password">Password</label>
            <input placeholder='Password' type="text" name='password' id='password' />
          </div>
          <button type='button' onClick={() => this.demoLogin('company')}>Sign In Company</button>
          <button type='button' onClick={() => this.demoLogin('volunteer')}>Sign In Volunteer</button>
          <p>Don't have an account? <Link to='/register'>Click Here</Link> to create one</p>
        </form>
      </div>
    )
  }
}