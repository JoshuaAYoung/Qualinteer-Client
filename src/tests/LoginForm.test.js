import React from 'react';
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import LoginForm from '../components/LoginForm/LoginForm'
import { BrowserRouter } from 'react-router-dom'

describe('LoginForm Component', () => {

  it('LoginForm Smoke Test', () => {
    const div = document.createElement('div')

    ReactDOM.render(<BrowserRouter><LoginForm /></BrowserRouter>, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('LoginForm Snapshot Test', () => {
    const wrapper = shallow(
      <LoginForm />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})