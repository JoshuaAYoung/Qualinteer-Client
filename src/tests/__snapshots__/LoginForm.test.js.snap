// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm Component LoginForm Snapshot Test 1`] = `
<div>
  <header
    className="loginBanner"
    role="banner"
  >
    <h1
      className="pageHeader"
    >
      Login.
    </h1>
    <p
      className="pageInstructions"
    >
      Enter your email and password or choose a demo user.
    </p>
  </header>
  <div
    role="alert"
  />
  <div
    className="demoInstructions"
  >
    <p
      className="defaultMedium"
    >
      Volunteer Demo:
    </p>
    <p
      className="demoLogin"
    >
      email@email.com / Joshs4pp
    </p>
    <p
      className="defaultMedium"
    >
      Organization Demo:
    </p>
    <p
      className="demoLogin"
    >
      j@young.net / P4ssword
    </p>
  </div>
  <form
    className="loginForm"
    onSubmit={[Function]}
  >
    <div
      className="inputContainer"
    >
      <input
        className="loginInput"
        id="email"
        name="email"
        onChange={[Function]}
        placeholder="Email"
        required={true}
        type="text"
      />
    </div>
    <div
      className="inputContainer"
    >
      <input
        className="loginInput"
        id="password"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        required={true}
        type="password"
      />
      <button
        className="showLoginButton"
        onClick={[Function]}
        type="button"
      />
    </div>
    <button
      className="loginButton"
      type="submit"
    >
      Sign In
    </button>
    <p
      className="registerInstructions"
    >
      Need an account? 
      <Link
        className="registerLink"
        to="/register"
      >
        Click Here
      </Link>
       to create one.
    </p>
  </form>
</div>
`;

exports[`NavBar Component NavBar Snapshot Test 1`] = `
<div
  className="navContainer"
>
  <div
    className="navBar"
  >
    <NavLink
      className="logoLink"
      onClick={[Function]}
      to="/"
    >
      <img
        alt="v-able logo"
        className="mainLogo"
        src="/assets/v-able-logo.svg"
      />
      <h1
        className="navName"
      >
        v
        <span
          className="inlineBullet"
        >
          â€¢
        </span>
        able
      </h1>
    </NavLink>
    <div
      className="hamburgerContainer"
    >
      <div
        className="burgerButton falseBurger"
        onClick={[Function]}
      >
        <span />
        <span />
        <span />
      </div>
    </div>
  </div>
  <div
    className="navMenu fadeMenuHidden"
  >
    <div
      className="linkContainer"
    >
      <NavLink
        className="navLink"
        onClick={[Function]}
        to="/login"
      >
        Log In.
      </NavLink>
    </div>
    <div
      className="linkContainer"
    >
      <NavLink
        className="navLink"
        onClick={[Function]}
        to="/register"
      >
        Sign Up.
      </NavLink>
    </div>
    <div
      className="linkContainer"
    >
      <NavLink
        className="navLink"
        onClick={[Function]}
        to="/opportunityboard"
      >
        Opportunity Board.
      </NavLink>
    </div>
  </div>
</div>
`;
